### PR TITLE
Fixes #2383 : Return ErrPartialViewErrors if any partial errors are present from a view

### LINKED ISSUE #2383: Channel cache missing data when request instantiating cache times out
Details here:
https://forums.couchbase.com/t/different-sync-gateway-instances-working-with-same-cluster-return-different-results/12086/7

Need additional investigation, but sounds like there's a potential problem with channel cache initialization when either the view query times out, or the changes request triggering the creation of the cache times out.

### PR DESCRIPTION
**Pull request by bbrks**
Fixes #2383 : Prevents the channel cache from being initilized in an invalid state 

# Changes

- Added test to repro #2383 

- `CouchbaseBucketGoCB.ViewCustom` and `CouchbaseBucketGoCB.View` now return `ErrPartialViewErrors` if any partial errors are present.

- Added the ability to use a `LeakyBucket` in `RestTester` by passing in a `LeakyBucketConfig`.

- Increased drone `test.sh -race` timeout to 20 minutes (from 10m) now we seem to be regularly hitting that limit (currently ~630s on drone)

# Integration Tests

- http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/284/
- http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-integration-master/285/

### CODE REVIEW

File: base/bucket_gocb.go [hunk 1]
```diff
@@ -1975,6 +1978,12 @@ func (bucket CouchbaseBucketGoCB) ViewCustom(ddoc, name string, params map[strin
 		return err
 	}
 
+	// Checking and returning down here as we still need to
+	// marshal/unmarshal into vres with partial errors present.
+	if errClose != nil {
```
Comment (adamcfraser) [inline base/bucket_gocb.go]: It might be more readable/maintainable to switch this to `if len(viewResponse.Errors) > 0` - what do you think?

File: base/bucket_gocb.go [hunk 2]
```diff
@@ -23,11 +23,12 @@ import (
 	"strconv"
 	"strings"
 
+	"sync"
```
Comment (adamcfraser) [inline base/bucket_gocb.go]: The empty line above sync should be removed (I realize sync was already misplaced in the imports before this PR).

File: .drone.yml [hunk 3]
```diff
@@ -10,4 +10,4 @@ build:
     - ./bootstrap.sh -c $$COMMIT -p sg
     - ./build.sh -v
     - ./test.sh
-    - ./test.sh -race
+    - ./test.sh -race -timeout 20m
```
Comment (adamcfraser) [inline .drone.yml:13]: Is this change required for the new test in this PR, or a change that's just generally needed?

**Review by adamcfraser — COMMENTED** [review]
Looks great - just a few minor comments/questions.

File: .drone.yml (hunk 3 above)
Comment (bbrks) [reply .drone.yml:13]: @adamcfraser It's not a requirement for the test included in this PR, but something is causing us to just tip over the default 10m timeout on drone. Could be a combination of things. I'll pull this out into a separate PR if nessesary.

File: base/bucket_gocb.go (hunk 1 above)
Comment (bbrks) [reply base/bucket_gocb.go]: @adamcfraser Yep, agreed.

**Review by adamcfraser — APPROVED** [review]
